README.md

```diff
@@ -40,57 +40,65 @@ To deploy Javabuilder to production, see the
-1. Click connect
-
-You should now start seeing messages from Javabuilder! Your token will last for 15
-minutes. If you make changes to the code, you do not need to re-generate your token.
-
-**Developing with deployed Javabuilder and an adhoc environment**
-1. Deploy Javabuilder with the instructions above
+There are two main ways to develop and run Javabuilder:
+- Using the **[WebSocketServer](https://github.com/code-dot-org/javabuilder/blob/main/org-code-javabuilder/lib/src/main/java/dev/javabuilder/WebSocketServer.java)**: 
+  - this is a local replacement of API Gateway that runs the "core" Javabuilder (the engine that 
+    builds & runs student code) directly. This suits most local development needs that don't touch 
+    any AWS-specific classes, or code outside the [org-code-javabuilder](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder) 
+    directory. To set up the WebSocketServer, skip directly to the
+    [org-code-javabuilder README](https://github.com/code-dot-org/javabuilder/blob/main/org-code-javabuilder/README.md).
+- Deploying a **development instance** of Javabuilder: 
+  - this allows you to develop against a custom deployment of the full Javabuilder AWS stack in a 
+    prod-like environment. This is useful for developing AWS-specific code (anything prefixed with 
+    `AWS` or `Lambda`), and anything outside the [org-code-javabuilder](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder) 
+    directory, such as the Lambda authorizers and deployment scripts. To deploy a development instance of 
+    Javabuilder, read further.
+
+### Deploying a Dev Instance
+
+1. Make and commit your desired changes.
+1. Deploy a development instance of Javabuilder, following the instructions here:
+   [Deploying Javabuilder](https://docs.google.com/document/d/1mMQK6HhniLsz9lynzhUcm7Tcw_2WVLBxADe0WzqL6rM/edit#bookmark=id.6objek4aiiu5).
+   The deployment should take around 10 minutes.
+
+To connect your dev instance with Java Lab (Code Studio client) running on your local Dashboard server:   
+
 1. In the code-dot-org repo, edit the `javabuilder_url` value in
-   [cdo.rb](https://github.com/code-dot-org/code-dot-org/blob/3219e5866689117e086d9891effe0fb39b9ae3f0/lib/cdo.rb#L131)
-   to point to your local dev deployment.
-1. In the code-dot-org repo, also edit the `upgrade_insecure_requests` list to include
-   the deployed Javabuilder hostname. 
+   [cdo.rb](https://github.com/code-dot-org/code-dot-org/blob/665a45210d556b4c3d82d6ad2434617c8e2e5ea1/lib/cdo.rb#L127)
+   to point to your local dev deployment (this will typically be wss://javabuilder-\<your name\>.dev-code.org).
+1. Edit the `javabuilder_upload_url` value in
+   [cdo.rb](https://github.com/code-dot-org/code-dot-org/blob/665a45210d556b4c3d82d6ad2434617c8e2e5ea1/lib/cdo.rb#L137)
+   to point to your local dev HTTP upload API (this will typically be https://javabuilder-\<your name\>-http.dev-code.org/seedsources/sources.json).
```

Comment:
It looks like the escape character is showing up in the rich diff `\<your name` - Not sure if that's just a fluke in the github diff or if it'll continue to show up after merged.

Reply:
Ah good catch - looks like for some reason this doesn't show up in the IntelliJ markdown viewer, but it does here and on other markdown viewers online? I'll remove it.